addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxNotificationPlayer.java

```diff
@@ -227,16 +227,20 @@ private void restorePlayerState() {
 
         // Resume playing save playlist item if player wasn't stopped.
         // Note that setting the time doesn't work for remote streams.
-        if (!playerState.isStopped()) {
-            logger.debug("Resuming last item playing");
-            squeezeBoxServerHandler.playPlaylistItem(mac, playerState.getPlaylistIndex());
-            squeezeBoxServerHandler.setPlayingTime(mac, playerState.getPlayingTime());
-        } else if (!playerState.isPlaying()) {
-            logger.debug("Pausing the player");
-            squeezeBoxServerHandler.pause(mac);
-        } else {
-            logger.debug("Stopping the player");
-            squeezeBoxServerHandler.stop(mac);
+        switch (playerState.getPlayState()) {
+            case PLAY:
+                logger.debug("Resuming last item playing");
+                squeezeBoxServerHandler.playPlaylistItem(mac, playerState.getPlaylistIndex());
+                squeezeBoxServerHandler.setPlayingTime(mac, playerState.getPlayingTime());
+                break;
+            case PAUSE:
```

Comment:
If the player is paused, I think it needs to have the saved playlist item and playing time restored. Since we just deleted the last item in the playlist (i.e. the notification), the LMS might be pointed at the last item in the playlist. As I recall, this might've been the scenario that was giving me fits, and led to the hack that sent pause multiple times. :-(